bin/cron/update_dotd

```diff
@@ -1,32 +1,50 @@
 #!/usr/bin/env ruby
+
 require 'open-uri'
 require 'json'
 require File.expand_path('../../../pegasus/src/env', __FILE__)
 require_relative '../../deployment'
 
+SLACK_TOKEN = CDO.SLACK_TOKEN.freeze
+PAGERDUTY_TOKEN = CDO.PAGERDUTY_TOKEN.freeze
+CHANNEL_ID = CDO.developers_CHANNEL_ID.freeze
+
+# @param user [Hash] PagerDuty information for a user
+# @return [Boolean] whether the user is the current DOTD primary oncall.
+def dotd?(user)
+  return false unless user['on_call']
+  user['on_call'].each do |on_call|
+    if on_call['level'] == 1 &&
+      on_call['escalation_policy']['name'] == 'Dev of the day issues'
+      return true
+    end
+  end
+  return false
+end
+
 def main
   return unless only_one_running?(__FILE__)
 
-  slack_token = CDO.slack_token
-  pagerduty_token = CDO.pagerduty_token
-  channel_id = CDO.developers_channel_id
-
-  # Get current developers room topic
-  channel_info = open("https://slack.com/api/channels.info?token=#{slack_token}&channel=#{channel_id}").read
+  # Get current developers room topic.
+  channel_info = open("https://slack.com/api/channels.info?token=#{SLACK_TOKEN}&channel=#{CHANNEL_ID}").read
   current_topic = JSON.parse(channel_info)["channel"]["topic"]["value"]
 
-  # Get DOTD on call from Pager Duty
-  on_call = `curl -H "Content-type: application/json" -H "Authorization: Token token=#{pagerduty_token}" -X GET -G "https://codeorg.pagerduty.com/api/v1/users/on_call"`
-  dotd_email = JSON.parse(on_call)["users"].first["email"]
+  # Get DOTD on call from PagerDuty.
+  on_call = `curl -H "Content-type: application/json" -H "Authorization: Token token=#{PAGERDUTY_TOKEN}" -X GET -G "https://codeorg.pagerduty.com/api/v1/users/on_call"`
+  on_call_parsed = JSON.parse(on_call)
+  dotd_email = nil
+  on_call_parsed['users'].each do |user|
+    dotd_email = user["email"] if dotd?(user)
+  end
 
-  # Match Pager Duty email to Slack mention name
-  users_list = open("https://slack.com/api/users.list?token=#{slack_token}").read
+  # Match PagerDuty email to Slack mention name.
+  users_list = open("https://slack.com/api/users.list?token=#{SLACK_TOKEN}").read
   dotd_user = JSON.parse(users_list)["members"].find { |member| dotd_email == member["profile"]["email"] }
```

Comment:
Note that if no on-call is found, then `dotd_email` will be `nil`. And this should make `dotd_user` also `nil`, giving an appropriate `dotd_mention`.